### ISSUE TITLE
Stage2/3 Finetunning

### ISSUE THREAD

**Issue by mxgx110** [hiyamdebary/EarthDial#4]
Hello, and thank you for the work and the shared code.

I’ve completed the first pretraining stage and have my models saved in `output_dir`. As I proceed to the second stage, I have a few questions regarding the setup, specifically related to the `EarthDial.sh` script:

1. In the `EarthDial.sh` script, the `--model_name_or_path` argument is set to `/pretrained/InternVL2-4B`, which appears to be a local directory. Only for making the training script work, I changed it to the Hugging Face model hub path: `OpenGVLab/InternVL2-4B` and it worked. Despite this and based on the training workflow, I assume this argument should instead point to the model checkpoint from Stage 1 (i.e., `output_dir`). Am I correct in this assumption? If not, which argument should be used to load the model from `Stage 1`?

2. In the appendix of the paper, it’s stated that for LoRA fine-tuning, a `LoRA rank` of 128 was used, with a `batch size` of 2 and a `learning rate` of 4e-5. It also mentions that ViT, and LLM components were frozen. Based on this, I expected the arguments to look like:

--freeze_llm True \
--use_llm_lora 128 \
--freeze_mlp False \
--freeze_backbone True \
--use_backbone_lora 128

However, the actual script contains:

--freeze_llm False \
--freeze_mlp False \
--freeze_backbone True

Could you clarify this?

3. Regarding the third stage, is there an existing `.sh` script provided for it, or should `EarthDial.sh` be modified accordingly to handle that stage?

Thank you in advance for your help!

**Comment by mustansarfiaz** [comment]
Hi

Thank you for being so interested, and it is a good question.
About (1), your assumptions are correct. After stage 1 finetuning, you should modify .sh accordingly and load weights from the local path.
About (2), for stage 3 fine-tuning, you should modify your .sh accordingly.

Regards
Mustansar

**Comment by SuiFengLai** [comment]
Hello, may I ask what your training script is in the third stage? I will always appear during the third stage of training.
warning: The size of tensor a (128) must match the size of tensor b (512) at non-singleton dimension 0, input_embeds[selected].shape=torch.Size([128, 3072]), vit_embeds.shape=torch.Size([512, 3072])，
